Sources/SwiftParser/Types.swift

```diff
@@ -224,13 +224,23 @@ extension Parser {
   ///     member-type-identifier-base → simple-type | member-type-identifier
   @_spi(RawSyntax)
   public mutating func parseSimpleType(
-    stopAtFirstPeriod: Bool = false
+    stopAtFirstPeriod: Bool = false,
+    isRawKeyPathExprSyntax: Bool = false
   ) -> RawTypeSyntax {
     var base: RawTypeSyntax
     switch self.currentToken {
+    case TokenSpec(.identifier):
+      var lookahead = self.lookahead()
+      lookahead.consumeAnyToken()
+      if !isRawKeyPathExprSyntax,
+         let _ = lookahead.consume(if: .leftSquareBracket, followedBy: .integerLiteral),
+         lookahead.currentToken.rawTokenKind == .rightSquareBracket {
+        base = RawTypeSyntax(self.parseCollectionType())
+      } else {
+        base = self.parseTypeIdentifier()
+      }
```

Comment:
Hmm, this is some non-trivial lookahead overhead that we need to pay for *every* type and the diagnostic is really not that important. Swift has been spelling arrays as `[Int]` instead of `Int[]` since beta 2 or 3 of Swift 1.0 C-programmers will stumble across bigger differences in the Swift language than the spelling of array types. With the current implementation, I think the performance cost is just not worth the benefit. 

I think either we find a solution to emit this diagnostic without doing lookahead or we just ditch it (i.e. just remove the TODO). And I wouldn’t be terribly opposed to just ditching it. I’ll leave it up to you how much effort you want to put into finding a non-lookahead solution.

Reply:
Having thought about it quite a bit, I believe that a non-lookahead solution is beyond my capability. The only thing i cold find was peeking for `[` in order to reduce the number of lookaheads. Do you want me to remove the TODOs?

Reply:
Yes, I think just removing the TODOs is reasonable.